Dfe.Data.SearchPrototype/Web/Tests/Acceptance/Steps/AccessibilitySteps.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Deque.AxeCore.Selenium;
+using DfE.Data.SearchPrototype.Web.Tests.Integration.PageObjectModel;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using OpenQA.Selenium;
+using TechTalk.SpecFlow;
+using Xunit.Abstractions;
+
+namespace Dfe.Data.SearchPrototype.Web.Tests.Acceptance.Steps
+{
+    [Binding]
+    public sealed class AccessibilitySteps
+    {
+        private readonly HomePage _homePage;
+        private readonly IWebDriver _driver;
+        private readonly ITestOutputHelper _logger;
+        private readonly ScenarioContext _scenarioContext;
+        
+        private Dictionary<string, string> _pageNameToUrlConverter = new Dictionary<string, string>()
+        {
+            { "home", "/" },
+            { "privacy", "/Home/Privacy" }
+        };
+
+        public AccessibilitySteps(
+            HomePage homePage,
+            IWebDriver driver,
+            ITestOutputHelper logger,
+            ScenarioContext scenarioContext
+        )
+        {
+            _driver = driver;
+            _homePage = homePage; 
+            _logger = logger;
+            _scenarioContext = scenarioContext;
+        }
+
+        [StepDefinition(@"the user views the (home|privacy) page")]
+        public void OpenPage(string pageName)
+        {
+            _driver.Navigate().GoToUrl($"http://localhost:5000{_pageNameToUrlConverter[pageName]}");
+            _homePage.Heading.Criteria.Should().NotBeNull();
+        }
+
+        [StepDefinition(@"the (.*) is accessible")]
+        public void IsAccessible(string component)
+        {
+            // see https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#axe-core-tags
+
+            var axeResult = new AxeBuilder(_driver)
```

Comment:
I think you asked if there was a way to run these tests without running the web app and using selenium. The answer is I don't know, but to find out, you would look to see whether there are other ways to construct this AxeBuilder class using maybe just the DOM

Reply:
Thanks. We're going with an agreed approach across the board which is Axe. At present we need Selenium to run it but it's something we can look into in the future.
https://www.gov.uk/service-manual/helping-people-to-use-your-service/testing-for-accessibility